### ISSUE TITLE
`ordered-imports` fails when `named-imports-order` is "any"

### ISSUE THREAD

**Issue by elwayman02** [palantir/tslint#1737]
### Bug Report

- __TSLint version__: `3.15.1`
- __TypeScript version__: `2.1.1`
- __Running TSLint via__: CLI

#### TypeScript code being linted with `tslint.json` configuration:

```json
"ordered-imports": [
      true, {
        "import-sources-order": "case-insensitive",
        "named-imports-order": "any"
      }
    ],
```

#### Actual behavior

```
 if (transform(xs[i].getText()) < transform(xs[i - 1].getText())) {
            ^

TypeError: transform is not a function
    at findUnsortedPair
```

#### Expected behavior

Linter should show violations

**Comment by elwayman02** [comment]
If I change the "any" to "case-insensitive", the error stops occurring.

**Comment by adidahiya** [comment]
yikes; thanks for the bug report!

**Comment by nchen63** [comment]
I can't reproduce this error in 4.0

**Comment by nchen63** [comment]
The `any` setting was added in 4.0.0-dev.1 https://github.com/palantir/tslint/commit/39ba6ade68a4a5d35332acb09e40d9546ce0eba8. We should do a better job of validating options, though
